File: Frontend/src/services/reportService.ts
import api from '../api/axios'
import { StandardResponse } from '../types'

export const reportService = {
  getDaily: async (storeId: number, date?: string) =>
    (await api.get<StandardResponse<Record<string, unknown>>>(`/api/reports/store/${storeId}/daily`, { params: { date } })).data.data,

  getMonthly: async (storeId: number, year?: number, month?: number) =>
    (await api.get<StandardResponse<Record<string, unknown>>>(`/api/reports/store/${storeId}/monthly`, { params: { year, month } })).data.data,

  getProductReport: async (storeId: number, from?: string, to?: string) =>
    (await api.get<StandardResponse<Record<string, unknown>>>(`/api/reports/store/${storeId}/products`, { params: { from, to } })).data.data,

  getInventoryReport: async (storeId: number) =>
    (await api.get<StandardResponse<Record<string, unknown>>>(`/api/reports/store/${storeId}/inventory`)).data.data,

  getChartData: async (storeId: number, days = 30) =>
    (await api.get<StandardResponse<Array<Record<string, unknown>>>>(`/api/reports/store/${storeId}/chart`, { params: { days } })).data.data,
}

File: Backend/BillX-Pos-System/src/main/java/com/BillX/Service/ReportService.java
package com.BillX.Service;

import java.time.LocalDate;
import java.util.List;
import java.util.Map;


public interface ReportService {
    Map<String, Object> getDailySalesReport(Long storeId, LocalDate date);
    Map<String, Object> getMonthlySalesReport(Long storeId, int year, int month);
    Map<String, Object> getProductSalesReport(Long storeId, LocalDate from, LocalDate to);
    Map<String, Object> getInventoryReport(Long storeId);
    List<Map<String, Object>> getSalesChartData(Long storeId, int days);
}
